ui/src/components/home/MailingListForm.js

```diff
@@ -0,0 +1,163 @@
+import {
+  Box,
+  Button,
+  Link,
+  Modal,
+  ModalCloseButton,
+  Image,
+  ModalContent,
+  AspectRatio,
+  ModalOverlay,
+  ModalHeader,
+  ModalBody,
+  ModalFooter,
+  VStack,
+  Text,
+  Input,
+  ButtonGroup,
+  useToast,
+} from '@chakra-ui/react';
+import { useState, useEffect } from 'react';
+import { useForm } from 'react-hook-form';
+import { customAxios as axios } from '../helpers/customAxios';
+
+const MailingListForm = ({ props }) => {
+  const user = props;
+  const [isOpen, setIsOpen] = useState(true);
```

Comment:
can we make the default state check if the user has ever submitted to this haha, was noticing that this modal will always popup no matter what if there's no other modal present. thinking a better UX would be they just need to submit once and then it will disappear. also prevents multi submission

Reply:
added, also added a date check to only show when date is before dec 1